### PR TITLE
Make bots.sh print failure for when it's run locally

### PR DESCRIPTION
**Pull request by tvolkert**

### CODE REVIEW

File: dev/bots/test.sh [hunk 1]
```diff
@@ -3,12 +3,18 @@ set -ex
 
 export PATH="$PWD/bin:$PWD/bin/cache/dart-sdk/bin:$PATH"
 
+die() {
+    set +x
+    echo "Error: script executed early due to error"
```
Comment (abarth) [inline dev/bots/test.sh]: exited?

Comment (tvolkert) [inline dev/bots/test.sh]: Done

**Comment by Hixie** [comment]
LGTM.

Does bash have a way to just do this kind of thing arbitrarily? It would be nice to show this message regardless.

Maybe wrap the whole file in a `(...) || echo 'Failed' && exit 1` or something?

**Comment by tvolkert** [comment]
Presumably the caller will see the error messages in the other parts of this file - the problem with these two is the stdout redirection to /dev/null (intentional -- we don't want to clutter the user's output with smoke tests)

**Comment by Hixie** [comment]
It's always possible for the other scripts to fail silently as well. :-)

**Comment by tvolkert** [comment]
Turns out there is a way to do what you want.  https://github.com/flutter/flutter/pull/4682 is a better version.

**Pull request merged**